### PR TITLE
Fix build issues and buffer-clearing issue for non-buffered video modes

### PR DESCRIPTION
**Pull request by Cwiiis**
This fixes building on Linux/Android for me, as well as fixing a glitch introduced in 2.8.2 where the background won't clear unless you pick native rendering resolution.

### CODE REVIEW

**Comment by blisstik** [comment]
I tried your fork, it does fix the buffer not clearing but reintroduces random flickering. Do you get random flickering? It's apparent in Die Hard for example.

File: yabause/src/vidogl.c [hunk 1]
```diff
@@ -4054,7 +4054,11 @@ int VIDOGLInit(void)
 
   SetSaturnResolution(320, 224);
 
+#if defined(YAB_ASYNC_RENDERING)
```
[the remark this answers is not in the document]
Comment (Cwiiis) [reply yabause/src/vidogl.c]: Ah, I've conflated two different options, I'll remove this commit.

File: yabause/src/android/jni/BackupManager.cpp [hunk 2]
```diff
@@ -97,7 +97,7 @@ int BackupManager::getDevicelist( string & jsonstr ) {
   }
 
   convJsontoString(root,jsonstr);
-  __android_log_print(ANDROID_LOG_INFO,LOGTAG,jsonstr.c_str());
+  __android_log_print(ANDROID_LOG_INFO,LOGTAG,"%s",jsonstr.c_str());
```
[the remark this answers is not in the document]
Comment (Cwiiis) [reply yabause/src/android/jni/BackupManager.cpp]: This line passes unescaped text into a formatted print function - this is a warning that stops the build on my setup (perhaps I'm using a more recent ndk). Warning aside, this is a potential security issue, so makes sense to do this.

File: yabause/CMake/Packages/external_libchdr.cmake [hunk 3]
```diff
@@ -42,6 +44,13 @@ set(LIBCHDR_LIBRARIES
  ${BINARY_DIR}/$<CONFIGURATION>/${CMAKE_STATIC_LIBRARY_PREFIX}crypto-static${CMAKE_STATIC_LIBRARY_SUFFIX}
  ${BINARY_DIR}/$<CONFIGURATION>/${CMAKE_STATIC_LIBRARY_PREFIX}lzma-static${CMAKE_STATIC_LIBRARY_SUFFIX}
 )
+elseif(UNIX AND NOT ANDROID)
+set(LIBCHDR_LIBRARIES  
+ ${BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}chdr-static${CMAKE_STATIC_LIBRARY_SUFFIX}
+ ${BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}flac-static${CMAKE_STATIC_LIBRARY_SUFFIX}
+ ${BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}crypto-static${CMAKE_STATIC_LIBRARY_SUFFIX}
+ ${BINARY_DIR}/${CMAKE_STATIC_LIBRARY_PREFIX}lzma-static${CMAKE_STATIC_LIBRARY_SUFFIX}
```
[the remark this answers is not in the document]
Comment (Cwiiis) [reply yabause/CMake/Packages/external_libchdr.cmake]: Without this change, linking fails building the Linux binary:

    [ 89%] Linking CXX executable yabasanshiro
    /usr/bin/ld: cannot find -lchdr-static
    /usr/bin/ld: cannot find -lflac-static
    /usr/bin/ld: cannot find -lcrypto-static
    /usr/bin/ld: cannot find -llzma-static
    /usr/bin/ld: cannot find -lchdr-static
    /usr/bin/ld: cannot find -lflac-static
    /usr/bin/ld: cannot find -lcrypto-static
    /usr/bin/ld: cannot find -llzma-static
    collect2: error: ld returned 1 exit status
    make[2]: *** [src/qt/CMakeFiles/yabause-qt.dir/build.make:1663: src/qt/yabasanshiro] Error 1
    make[1]: *** [CMakeFiles/Makefile2:369: src/qt/CMakeFiles/yabause-qt.dir/all] Error 2
    make: *** [Makefile:152: all] Error 2

**Comment by devmiyax** [comment]
Thank you for collaboration.
I've merged your code by hand only the part where I think it is needed.

https://github.com/devmiyax/yabause/commit/8cd09dce95cd7315e0af472548b4f8e70860a314

**Pull request closed without merging**